netsec-ethz/scion-coord
# New vagrant box

This PR automates the creation of the base box for vagrant.
This closes #44 
To test you have to:
- Create a new box using the script. Ensure you add the box locally (use the `-p` switch to the script)
- Download a VM from the coordinator, run it and check it is okay
- Stop the previous VM and edit its Vagrantfile. You will have to pin the version to 0.0.1.2 (that's the one using for testing in the script right now), by adding 
`config.vm.box_version = "0.0.1.2"` right below `config.vm.box = "scion/ubuntu-16.04-64-scion"`. This way vagrant creates the VM based on the testing base one.
- Copy the `scion_install_script.sh` file to the same folder where Vagrantfile is. Replace the call to it in the Vagrantfile with `/vagrant/scion_install_script.sh` instead, and don't download anything from github (remove wget, etc)
- now `vagrant up` the VM

Things to test:
- Older VMs still work when referencing this base box
- New ones as well
- Upgrade process is running correctly


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/netsec-ethz/scion-coord/262)
